libraries/AP_ESC_Telem/AP_ESC_Telem.cpp

```diff
@@ -288,13 +288,27 @@ void AP_ESC_Telem::update_telem_data(uint8_t esc_index, const AP_ESC_Telem_Backe
     }
 
     _telem_data[esc_index].count++;
+    if (_telem_data[esc_index].count == 0) {
+        // reset the CRC error counter each time the telemetry packet counter overflows,
+        // to prevent calculating CRC error rate with the incorrect (overflown) packet counter.
+        _telem_data[esc_index].CRC_error_counter = 0;
+    }
     _telem_data[esc_index].types |= data_mask;
     _telem_data[esc_index].last_update_ms = AP_HAL::millis();
 }
 
+// callback to increment the CRC error counter data in the frontend, should be called by the driver when a CRC error occurs
+void AP_ESC_Telem::increment_CRC_error_counter(const uint8_t esc_index)
+{
+    if (esc_index > ESC_TELEM_MAX_ESCS) {
+        return;
+    }
+    _telem_data[esc_index].CRC_error_counter++;
+}
+
 // record an update to the RPM together with timestamp, this allows the notch values to be slewed
 // this should be called by backends when new telemetry values are available
-void AP_ESC_Telem::update_rpm(uint8_t esc_index, uint16_t new_rpm, float error_rate)
```

Comment:
This is wrong, the error rate comes directly from bi-directional dshot not from the CRC. We could possibly merge the two if that's what the non-bidir dshot rate means

Reply:
There are multiple communication errors that can be detected and reported:
- CRC errors in AP to ESC communication (detected by the ESC)
- CRC errors in ESC to AP communication (detected by AP)
- Bi-DShot errors coming from the ESC ???? (detected by the ESC ???)

This API is meant for the second type of error. I assumed the old API was also for the second type of error. Can you elaborate a bit more on that third type of error?